### PR TITLE
DOC: Add instructions for contributing to Cartopy

### LINKED ISSUE #1371: Document test procedure
I find myself forgetting exactly what commands to run for the pytest test suite here. I think some kind of testing section in the readme just giving a few commands on how to test would be very useful.

I was running
`pytest --pyargs cartopy.tests`
or
`pytest --pyargs cartopy.tests.specific_test_file.py`
which I'm not even entirely sure is the best/right way to run the tests.

For example from matplotlib:
https://github.com/matplotlib/matplotlib#test
Maybe even a developer's guide page if you want to get really detailed?
https://matplotlib.org/devel/coding_guide.html#testing

### PR DESCRIPTION
**Pull request by greglucas**
Describes how to set up a development environment and outlines the steps required for contributing code.

Closes #1371

### CODE REVIEW

File: docs/source/contribute.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+to start contributing is by fixing typos, adding examples, or improving the explanations
+of the functions.
+
+To contribute new features or bug fixes, you can create a pull request with your code changes
+on GitHub. The easiest way to get started contributing code is to clone the repository to your
+local system and install the conda development environment included with the source code.
+
+.. code::
+
+  git clone git@github.com:YOUR-USERNAME/cartopy.git
```
Comment (dopplershift) [inline docs/source/contribute.rst:16]: Do we need to explain how to fork to their account above first?

Comment (greglucas) [reply docs/source/contribute.rst:19]: I'm not sure what the best way to do that is? I added text and a link to `/scitools/cartopy/fork`, which I think will get people started...

File: docs/source/contribute.rst [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+.. code::
+
+  # Activate your development environment
+  conda activate cartopy-dev
+
+  # Checkout a new feature branch
+  git checkout -b my-cool-feature
+
+  # Make your code changes here, including tests for the changes!
+  # add and commit the changes with a useful commit message
```
Comment (QuLogic) [inline docs/source/contribute.rst:43]:
```suggestion
  # Add and commit the changes with a useful commit message
```

File: docs/source/contribute.rst [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  # describing what changes you made
+  git add lib/cartopy/my-changed-file.py
+  git commit
+
+  # Test your code on 4 processors
+  pytest -n 4 lib/cartopy
+
+  # push your changes to your GitHub account
+  git push --set-upstream origin my-coll-feature
```
Comment (QuLogic) [inline docs/source/contribute.rst:52]:
```suggestion
  git push --set-upstream origin my-cool-feature
```

File: docs/source/contribute.rst [hunk 4]
```diff
@@ -0,0 +1,54 @@
+.. _contribute:
+
+Contributing to Cartopy
+=======================
+
+Cartopy is an open-source project that welcomes contributions from anyone. A good place
+to start contributing is by fixing typos, adding examples, or improving the explanations
+of the functions.
```
Comment (QuLogic) [inline docs/source/contribute.rst:8]: Until #1919 is resolved, the CLA should probably be mentioned up front.

Comment (greglucas) [reply docs/source/contribute.rst]: Good point, but I'm not sure how much detail we really want to go into here as I find the more text we start including the less likely people are to contribute! What about this as a compromise on more info, but not scaring people away?

"More information about the copyright and license agreement can be found on the `SciTools Governance page <https://scitools.org.uk/organisation.html>`_."

**Review by QuLogic — APPROVED** [review]

**Pull request merged**